pkg/registry/apis/iam/team_binding_hooks.go

```diff
@@ -159,35 +120,25 @@ func (b *IdentityAccessManagementAPIBuilder) BeginTeamBindingUpdate(ctx context.
 		return nil, nil
 	}
 
-	// Convert old team binding to tuple for deletion
-	var oldTuple *v1.TupleKey
-	var oldErr error
-	if oldTB.Spec.Subject.Name != "" && oldTB.Spec.TeamRef.Name != "" {
-		oldTuple, oldErr = convertTeamBindingToTuple(oldTB)
-		if oldErr != nil {
-			b.logger.Error("failed to convert old team binding to tuple",
-				"namespace", oldTB.Namespace,
-				"name", oldTB.Name,
-				"err", oldErr,
-			)
-			return nil, nil
-		}
-	}
-
-	// Convert new team binding to tuple for writing
-	var newTuple *v1.TupleKey
-	var newErr error
-	if newTB.Spec.Subject.Name != "" && newTB.Spec.TeamRef.Name != "" {
-		newTuple, newErr = convertTeamBindingToTuple(newTB)
-		if newErr != nil {
-			b.logger.Error("failed to convert new team binding to tuple",
-				"namespace", newTB.Namespace,
-				"name", newTB.Name,
-				"err", newErr,
-			)
-			return nil, nil
-		}
-	}
+	operations := make([]*v1.MutateOperation, 0, 2)
+	operations = append(operations, &v1.MutateOperation{
+		Operation: &v1.MutateOperation_DeleteTeamBinding{
+			DeleteTeamBinding: &v1.DeleteTeamBindingOperation{
+				SubjectName: oldTB.Spec.Subject.Name,
+				TeamName:    oldTB.Spec.TeamRef.Name,
+				Permission:  string(oldTB.Spec.Permission),
+			},
+		},
+	})
+	operations = append(operations, &v1.MutateOperation{
+		Operation: &v1.MutateOperation_CreateTeamBinding{
+			CreateTeamBinding: &v1.CreateTeamBindingOperation{
+				SubjectName: newTB.Spec.Subject.Name,
+				TeamName:    newTB.Spec.TeamRef.Name,
+				Permission:  string(newTB.Spec.Permission),
+			},
+		},
+	})
```

Comment:
Suggestion: early return saves a go routine and the ztickets wait.
```suggestion
	})
	if len(operations) == 0 {
		return func(ctx context.Context, success bool) {}
	}
```